### PR TITLE
feat(events): Add fixation location event property

### PR DESCRIPTION
**Pull request by SiQube**
since I had the property locally, first attempt at an implmentation for the package. feedback welcome

### CODE REVIEW

File: src/pymovements/events/event_properties.py [hunk 1]
```diff
@@ -175,3 +175,85 @@ def _check_velocity_columns(velocity_columns: tuple[str, str]) -> None:
...
+    ValueError
+        If centroid calculation is not yet supported -- currently we are supporting mean and median
+        fixation location position.
+    """
+    _check_position_columns(position_columns)
+
+    x_position = pl.col(position_columns[0])
+    if centroid_calculation not in ['mean', 'median']:
+        raise ValueError(
+            f'Centroid calculation {centroid_calculation} not (yet) supported '
```
Comment (dkrako) [inline src/pymovements/events/event_properties.py:208]: drop the "(yet)"

File: tests/events/event_properties_test.py [hunk 2]
```diff
@@ -107,6 +107,20 @@
             ('position_columns', 'type', 'must', 'tuple', 'str', 'int'),
             id='amplitude_tuple_of_int',
         ),
+        pytest.param(
+            event_properties.fixation_centroid_x,
+            {'centroid_calculation': 'foo'},
+            ValueError,
+            ('centroid', 'foo', 'supported', 'mean', 'median'),
```
Comment (dkrako) [inline tests/events/event_properties_test.py:114]: use `'not supported'` instead

File: src/pymovements/events/event_properties.py [hunk 3]
```diff
@@ -175,3 +175,85 @@ def _check_velocity_columns(velocity_columns: tuple[str, str]) -> None:
             'velocity_columns must be of type tuple[str, str] but is '
             f'tuple[{type(velocity_columns[0]).__name__}, {type(velocity_columns[1]).__name__}]',
         )
+
+
+@register_event_property
+def fixation_centroid_x(
+        position_columns: tuple[str, str] = ('x_pos', 'y_pos'),
+        centroid_calculation: str = 'mean',
```
[the remark this answers is not in the document]
Comment (SiQube) [reply src/pymovements/events/event_properties.py:183]: done

File: src/pymovements/events/event_properties.py [hunk 4]
```diff
@@ -175,3 +175,85 @@ def _check_velocity_columns(velocity_columns: tuple[str, str]) -> None:
...
+) -> pl.Expr:
+    """Calculate a fixations x centroid.
+
+    Parameters
+    ----------
+    position_columns
+        The column names of the pitch and yaw position components.
+    centroid_calculation
+        Specify the type of centroid to be extracted, defaults to 'mean'.
```
[the remark this answers is not in the document]
Comment (SiQube) [reply src/pymovements/events/event_properties.py:192]: done

File: src/pymovements/events/event_properties.py [hunk 5]
```diff
@@ -175,3 +175,85 @@ def _check_velocity_columns(velocity_columns: tuple[str, str]) -> None:
...
+
+    Raises
+    ------
+    TypeError
+        If position_columns not of type tuple, position_columns not of length 2, or elements of
+        position_columns not of type str.
+    ValueError
+        If centroid calculation is not yet supported -- currently we are supporting mean and median
```
[the remark this answers is not in the document]
Comment (SiQube) [reply src/pymovements/events/event_properties.py:200]: done

File: src/pymovements/events/event_properties.py (hunk 1 above)
Comment (SiQube) [reply src/pymovements/events/event_properties.py:208]: done

File: src/pymovements/events/event_properties.py [hunk 6]
```diff
@@ -175,3 +175,85 @@ def _check_velocity_columns(velocity_columns: tuple[str, str]) -> None:
...
+            f"please choose one of the following: ['mean', 'median'].",
+        )
+
+    if centroid_calculation == 'mean':
+        ret = x_position.mean()
+
+    if centroid_calculation == 'median':
+        ret = x_position.median()
+
+    return ret
```
[the remark this answers is not in the document]
Comment (SiQube) [reply src/pymovements/events/event_properties.py:218]: done

File: tests/events/event_properties_test.py (hunk 2 above)
Comment (SiQube) [reply tests/events/event_properties_test.py:114]: done

**Comment by dkrako** [comment]
Thanks again for your work, looks great! 

But I think you have missed the review comment about merging the two functions using a new `component` argument.

Also, move the merged function above the two private module functions `_check_position_columns` and `_check_velocity_columns`.

**Comment by SiQube** [comment]
the column names stay the same then unfortunately, can you give me some pointers how I can change the column names for x/y?

**Comment by dkrako** [comment]
I'm not completely sure what you mean. Do you mean the output or the input columns?

You can get the input column by using the `component` argument as the index of the `position_columns` tuple.
The output columns will be then the same as the input columns actually.

But maybe you refer to the output columns in some test cases?
These are assigned using the `alias()` method, see for example here:
https://github.com/aeye-lab/pymovements/blob/9e175a3313eb604951e7ecf0659b70f63bc5bb6d/tests/events/event_properties_test.py#L536

I am still thinking about how to integrate this well with the coming `apply()` method so don't worry about the output column name for now.

**Comment by SiQube** [comment]
I've added it  -- but now we have just one column in the `result_df`, namely `fixation_centroid` and I don't know how to adjust it given the component.

**Comment by dkrako** [comment]
LGTM!

The issue with the output column names will be resolved by the implementation of the apply method.

**Pull request merged**